Source/Scene/WebMapTileServiceImageryProvider.js

```diff
@@ -372,6 +380,8 @@ function requestImage(imageryProvider, col, row, level, request, interval) {
       queryParameters: query,
       request: request,
     });
+    //When useKvp is true, replace the {s} keyword in the requested URL. QUESTIONS: modify `resource`? Should I replace {s} with ""?
+    resource.url = resource.url.replace("{s}", "");
```

Comment:
And speaking of specs, it'd be good to add a spec that tests for this new functionality. I think something similar to the `"generates expected tile urls"` spec with `{s}` in the url option and a few query parameters would be fine, just to make sure that the query doesn't get dropped when `{s}` is added to the url.

Reply:
I just added the spec `generates expected tile urls for subdomains` to `WebMapTileServiceImageryProviderSpec.js`. As you specified, it is similar to the spec `generates expected tile urls` yet has `url: "http://wmts{s}.invalid"`. I left the query parameters the same as in `generates expected tile urls` - is this what you intended? 

FYI `resource.url = resource.url.replace("{s}", "");` does not pass the specs `generates expected tile urls for subdomains`, `supports subdomains string urls`, and `supports subdomains array urls`.

Reply:
Why are we replacing `{s}` with nothing. Wouldn't we want this for KVP in case there is a `{s}` in the url?
```
 var templateValues = {
  s:  subdomains[(col + row + level) % subdomains.length],
 };
  resource.setTemplateValues(templateValues);
```

Reply:
I was thinking the same as Fili in my other review comment.

>  I am not completely sure where your code snippet should be added.

@srothst1 - I was thinking this could go before the call to `imageryProvider._resource.getDerivedResource()` in the else-block when `_useKvp` is true.